### PR TITLE
{Test} Fix resource group deletion for cross tenant tests

### PR DESCRIPTION
**Pull request by zhoxing-ms**
**Description**

Related PR: https://github.com/Azure/azure-cli/pull/19159
Use `@ResourceGroupPreparer(... ,subscription=cross_tenant_subscrption)` to ensure that the cross account resource group will be deleted after the test is completed.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 1]
```diff
@@ -3510,7 +3514,7 @@ def test_deployment_group_cross_tenant(self, resource_group):
             'version': '1.1.2',
             'captured': self.create_random_name('cli_crosstenantmanagedimage', 40),
             'aux_sub': '1c638cf4-608f-4ee6-b680-c329e824c3a8',
-            'rg': self.create_random_name('cli_test_cross_tenant_rg', 40),
+            'rg': another_resource_group,
             'aux_tenant': '72f988bf-86f1-41af-91ab-2d7cd011db47'
         })
         self.cmd('group create -g {rg} --location {location} --subscription {aux_sub}',
```
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:3520]: delete previous resource group create command?

**Review by kairu-ms — APPROVED** [review]

**Comment by yonzhan** [comment]
fix cross tenant tests

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_general_operations.py [hunk 2]
```diff
@@ -197,15 +197,17 @@ class MonitorCloneStorageAccountAcrossSubsScenarios(ScenarioTest):
...
             'sa': self.create_random_name('sa', 24),
             'sa2': self.create_random_name('sa', 24),
             'ag1': 'ag1',
             'rg': resource_group,
             'ext_sub': '1c638cf4-608f-4ee6-b680-c329e824c3a8',
-            'ext_rg': self.create_random_name('test_rg', 24),
+            'ext_rg': another_resource_group,
         })
 
         self.cmd('group create -l eastus -n {ext_rg} --subscription {ext_sub}')
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_general_operations.py:213]: Unnecessary to create rg again as you create it in `ResourceGroupPreparer`

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 3]
```diff
@@ -3387,7 +3389,7 @@ def test_group_deployment_cross_tenant(self, resource_group):
...
```
Comment (jsntcy) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:3395]: Same issue

**Comment by jsntcy** [comment]
>         self.cmd('group create -g {another_rg} --location {location} --subscription {aux_sub}',

same issue

---
Refers to: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:4180 in 7a0237b. [](commit_id = 7a0237bffabb27c79fc1412c9481aa9c7bf693fe, deletion_comment = False)

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:3520]: There was no commands to delete resources in this test before. I assume it depends on the deletion task under the subscription to delete those resources

File: src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_general_operations.py (hunk 2 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/monitor/tests/latest/test_monitor_general_operations.py:213]: OK~

**Review by jsntcy — APPROVED** [review]
:shipit:

**Review by Juliehzl — APPROVED** [review]
LGTM. a small suggestion is that we can make aux sub as global variable and used by all tests.

**Pull request merged**